README.md

```diff
@@ -187,9 +187,20 @@ they exist in the local directory:
   next to image files it boots, for additional container
   runtime settings.
 
-* `mkosi.cache` may be a directory. If so, it is automatically used as
+* `mkosi.cache/` may be a directory. If so, it is automatically used as
   package download cache, in order to speed repeated runs of the tool.
 
+* `mkosi.builddir/` may be a directory. If so, it is automatically
+  used as out-of-tree build directory, if the build commands in the
+  `mkosi.build` script support it. Specifically, this directory will
+  be mounted into the build countainer, and the $BUILDDIR environment
+  variable will be set to it when the build script is invoked. The
+  build script may then use this directory as build directory, for
+  automake-style or ninja-style out-of-tree builds. This speeds up
+  builds considerably, in particular when `mkosi` is used in
+  incremental mode (`-i`): not only the disk images but also the build
+  tree is reused between subsequent invocations.
```

Comment:
As mentioned in the discussion, I think explicitly noting that `$BUILDDIR` will not be set if the dir is empty would be nice.